src/VisualStudio/Core/Def/DocumentOutline/DocumentOutlineControl.xaml.cs

```diff
@@ -0,0 +1,376 @@
+
+        public DocumentOutlineControl(
+            ILanguageServiceBrokerShim languageServiceBroker,
+            IThreadingContext threadingContext,
+            IAsynchronousOperationListener asyncListener,
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
+            IVsCodeWindow codeWindow)
+        {
+            InitializeComponent();
+
+            ThreadingContext = threadingContext;
+            LanguageServiceBroker = languageServiceBroker;
+            EditorAdaptersFactoryService = editorAdaptersFactoryService;
+            CodeWindow = codeWindow;
+            ComEventSink.Advise<IVsCodeWindowEvents>(codeWindow, this);
+            SortOption = SortOption.Order;
+
+            _getModelQueue = new AsyncBatchingWorkQueue(
+                    DelayTimeSpan.Short,
+                    GetModelAsync,
+                    asyncListener,
+                    threadingContext.DisposalToken);
+
+            _updateModelQueue = new AsyncBatchingWorkQueue(
+                    DelayTimeSpan.NearImmediate,
+                    UpdateModelAsync,
+                    asyncListener,
+                    threadingContext.DisposalToken);
+
+            _highlightNodeQueue = new AsyncBatchingWorkQueue(
+                    DelayTimeSpan.NearImmediate,
+                    HightlightNodeAsync,
+                    asyncListener,
+                    threadingContext.DisposalToken);
+
+            codeWindow.GetPrimaryView(out var pTextViewPrimary);
+            StartTrackingView(pTextViewPrimary, false, out var textViewPrimary);
+
+            if (textViewPrimary is null)
+                return;
+
+            codeWindow.GetSecondaryView(out var pTextViewSecondary);
+            if (pTextViewSecondary is not null)
+                StartTrackingView(pTextViewSecondary, false, out var _);
+
+            StartGetModelTask();
+        }
+
+        int IVsCodeWindowEvents.OnNewView(IVsTextView pView)
+        {
+            StartTrackingView(pView, true, out var _);
+            return VSConstants.S_OK;
+        }
+
+        private void StartTrackingView(IVsTextView pTextView, bool isNewView, out IWpfTextView? wpfTextView)
+        {
+            ThreadingContext.ThrowIfNotOnUIThread();
+            wpfTextView = null;
+            if (pTextView != null)
+            {
+                wpfTextView = EditorAdaptersFactoryService.GetWpfTextView(pTextView);
+
+                if (wpfTextView != null)
+                {
+                    _trackedTextViews.Add(pTextView, wpfTextView);
+                    wpfTextView.Caret.PositionChanged += Caret_PositionChanged;
+                    wpfTextView.TextBuffer.Changed += TextBuffer_Changed;
+
+                    // If textViewPrimary was null on initialization, we want to get the
+                    // document model once we obtain a text view
+                    if (isNewView && !DocumentSymbolViewModelsIsInitialized)
+                        StartGetModelTask();
```

Comment:
I'm not sure if textViewPrimary would ever be null on Document Outline window initialization (I haven't run into it while testing) so I figured I should handle it just in case (I'm reaching out to the editor team to confirm).

Reply:
The editor team has confirmed that it should never be null.